tox.ini

```diff
@@ -215,6 +215,13 @@ deps = .[modin-development]
        numpy<2.0.0
 commands = python -m pytest tests/integ/modin/test_old_numpy_aliases.py
 
+[testenv:datasource]
+description = run the datasource tests that requires DBMS drivers
+deps =
+    {[testenv]deps}
+    databricks-sql-connector
+commands = {env:SNOWFLAKE_PYTEST_CMD} -m "{env:SNOWFLAKE_TEST_TYPE}" {posargs:} tests/integ/datasource
```

Comment:
it seems like test_data_source_api.py is not in tests/integ/datasource, meaning the new pipeline will only run databricks test, is this expected?

Reply:
besides, these tests are not removed from other jobs, means they will be run multiple times, is that ok?

Reply:
my goal is
1. remove those drivers from our setup.py dev dependency
2. have a new tox env to install the dependencies we need for our tests
3. use a new tox env to run datasource tests that depend on those dependency

- first question:
yes, it's by design, we shall move test_data_source_api.py to tests/integ/datasource after using real test sources instead of the fake ones

- second question: other jobs will skip the tests due to lack of dependency